src/components/modules/tools.ts

```diff
@@ -248,9 +187,56 @@ export default class Tools extends Module {
         class: Stub,
         isInternal: true,
       },
+      moveUpTune: {
+        class: MoveUpTune,
+        isInternal: true,
+      },
+      deleteTune: {
+        class: DeleteTune,
+        isInternal: true,
+      },
+      moveDownTune: {
+        class: MoveDownTune,
+        isInternal: true,
+      },
     };
   }
 
+  /**
+   * Returns Block Tunes for passed Tool
+   *
+   * @param tool - Tool object
+   */
+  public getTunesForTool(tool: BlockTool): ToolsCollection<BlockTune> {
+    const names = tool.enabledBlockTunes;
+
+    if (names === false) {
+      return new ToolsCollection<BlockTune>();
+    }
+
+    if (Array.isArray(names)) {
+      return new ToolsCollection<BlockTune>(
+        Array
+          .from(this.blockTunes.entries())
+          .filter(([, tune]) => names.includes(tune.name))
+          .concat([ ...this.blockTunes.internalTools.entries() ])
+      );
+    }
+
+    const defaultTuneNames = this.config.tunes;
+
+    if (Array.isArray(defaultTuneNames)) {
+      return new ToolsCollection<BlockTune>(
```

Comment:
duplicated code fragments

Reply:
Let's keep it for now, I have idea to refactor it later